lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -729,6 +730,173 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
                                                         mulFOp.getOut());
 }
 
+LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
+                                     CmpFOp cmpf) const {
+  Location loc = cmpf.getLoc();
+  IntegerType one = rewriter.getI1Type(), five = rewriter.getIntegerType(5),
+              width = rewriter.getIntegerType(
+                  cmpf.getLhs().getType().getIntOrFloatBitWidth());
+  auto calyxCmpFOp = getState<ComponentLoweringState>()
+                         .getNewLibraryOpInstance<calyx::CompareFOpIEEE754>(
+                             rewriter, loc,
+                             {one, one, one, width, width, one, one, one, one,
+                              one, five, one});
+  hw::ConstantOp c1 = createConstant(loc, rewriter, getComponent(), 1, 1);
+  rewriter.setInsertionPointToStart(getComponent().getBodyBlock());
+
+  auto createSignalRegister =
```

Comment:
I'm not saying you need to have a function in CalyxLoweringUtil. I'm suggesting it should be a class method, for a few reasons. 

1. You have 9 arguments to this function. That's a lot, regardless if its a lambda call. Can't this be reduced by making it a method, e.g., just call `getComponent()` instead of have a `component` argument?
2. Modular code is good; that's why functions exist. We could obviously inline everything into a `main` function, but then it can be quite difficult to grasp the high level understanding of the program. 
3. I really believe lambdas should be short and sweet. If they're large, they should be contained in their own function. This is personal opinion though, hence why it is a nit.



(You can also put this locally as a static function. See: https://llvm.org/docs/CodingStandards.html#anonymous-namespaces)

Reply:
makes senses, just changed that!